### PR TITLE
Clarify behaviour of Rendezvous()

### LINKED ISSUE #584: Rendezvous() text should be a bit more detailed
See this email thread:  https://mailarchive.ietf.org/arch/msg/taps/B5iPOvreJ49aSINAqrM04jNNOVk/

Copy + pasting a relevant part from an email from @csperkins:

```
This is one of the areas where the drafts likely need expanding, but I think the model is:

1) application configures STUN (and maybe also TURN) local endpoints

2) application calls Resolve() on these, to find server reflexive candidates

3) application shares these candidates with its peer, via out-of-band means, and receives remote endpoint candidates from the peer

4) application adds remote endpoints for the peer’s candidates

5) application calls Rendezvous(), which performs the ICE-style probing to find a working path, based on the candidates, then returns s Connection

This relies on the Endpoint abstraction supporting STUN, which is maybe implicit in the drafts. 

```

To me at least, it wasn't clear from the text that rendezvous() would kick off ICE-style probing. It's kind of obvious (when else would it happen?), but I think it would be good for the draft to spell this out.

### PR DESCRIPTION
**Pull request by csperkins**
Closes #584

### CODE REVIEW

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -1413,9 +1413,42 @@ Remote Endpoint, and also the transport properties and security parameters
...
-Endpoint for an incoming Connection from the Remote Endpoint, while
-simultaneously trying to establish a Connection from the Local Endpoint to the
-Remote Endpoint. This corresponds to a TCP simultaneous open, for example.
+Endpoint for an incoming Connection from the Remote Endpoint, while also
+simultaneously trying to establish a Connection from the Local Endpoint to
+the Remote Endpoint. 
+
+If there are multiple local or remote endpoints configured, then a rendezvous
+action will systematically probe reachability of those endpoints following the
+approach used in Interactive Connectivity Establishment (ICE) {{?RFC5245}}.
```
Comment (britram) [inline draft-ietf-taps-interface.md:1422]: Do we need to specify this this tightly? (I think it's safe to do so, ICE is an approach, not a protocol. But we've pointedly avoided to make tight bindings to a Single Way Of Doing Things, so we should make this choice explicitly)

Comment (britram) [reply draft-ietf-taps-interface.md:1422]: also, if we're specifying ICE, this needs to be {{!RFC5245}}.

Comment (csperkins) [reply draft-ietf-taps-interface.md:1422]: Will rephrase to be slightly more general: "If there are multiple local or remote endpoints configured, then a rendezvous action will need to systematically probe reachability of those endpoints following an approach such as that used in Interactive Connectivity Establishment (ICE) {{?RFC5245}}."

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -1413,9 +1413,42 @@ Remote Endpoint, and also the transport properties and security parameters
...
 
 The Rendezvous() Action causes the Preconnection to listen on the Local
-Endpoint for an incoming Connection from the Remote Endpoint, while
-simultaneously trying to establish a Connection from the Local Endpoint to the
-Remote Endpoint. This corresponds to a TCP simultaneous open, for example.
+Endpoint for an incoming Connection from the Remote Endpoint, while also
+simultaneously trying to establish a Connection from the Local Endpoint to
+the Remote Endpoint. 
+
+If there are multiple local or remote endpoints configured, then a rendezvous
```
Comment (gorryfair) [inline draft-ietf-taps-interface.md:1420]: Should Local and Remote Endpoint be capitalised?

**Review by mwelzl — CHANGES_REQUESTED** [review]
Thanks a lot for doing this, this is what I was hoping to see!

So, this LGTM except that I also request what @britram already said - maybe you could just do a quick check of the whole text (not just the sentence @britram quoted) to see if it's too prescriptive.

**Comment by csperkins** [comment]
@mwelzl 7df2d9c should address this

**Comment by csperkins** [comment]
```
   The Preconnection Object must be specified with both a Local Endpoint
   and a Remote Endpoint, and also the transport properties and security
   parameters needed for Protocol Stack selection.
```

To `Rendezvous()` we need multiple local and remote candidates. Should this be represented as either:
1) `Rendezvous()` taking two lists of `Preconnection` objects; or
2) as two `Preconnection` objects each with multiple addresses, ports, etc.

I suggest option 1, since this makes it easier to resolve a `Preconnection` to multiple preconnections, for NAT binding discovery.

Relates to #571 but with also different types of endpoint.

**Comment by mwelzl** [comment]
I like option 1 because it contains complexity.

**Review by mwelzl — APPROVED** [review]

**Pull request merged**